frontend/src/components/interaction-manager/interaction-flow.tsx

```diff
@@ -60,7 +63,21 @@ const InteractionItem = (
     case INTERACTION_TYPE.REQUEST_TRANSACTION_REVIEW_FOR_SENDING: {
       return (
         <Transaction
-          {...(props as InteractionContentProps<RequestTransactionSending>)}
+          {...(props as InteractionContentProps<RequestTransactionReview>)}
```

Comment:
question (non blocking): any way we can improve the types here so as not to type cast? Perhaps a specific property to pass in specific interaction type data and/or lifting the props.interaction.event.name so you can discriminate the unions against it?

Reply:
i think destucturing ruins the type discrimination, so when i switched to spoon-feeding typescript with each prop explicitly, i didn't have to type cast anymore. it's a bit more verbose this way, but typescript's happier.